src/gil.rs

```diff
@@ -139,25 +135,9 @@ where
 }
 
 /// RAII type that represents the Global Interpreter Lock acquisition.
-///
-/// Users are strongly encouraged to use [`Python::with_gil`](struct.Python.html#method.with_gil)
-/// instead of directly constructing this type.
-/// See [`Python::acquire_gil`](struct.Python.html#method.acquire_gil) for more.
-///
-/// # Examples
-/// ```
-/// use pyo3::Python;
-///
-/// {
-///     #[allow(deprecated)]
-///     let gil_guard = Python::acquire_gil();
-///     let py = gil_guard.python();
-/// } // GIL is released when gil_guard is dropped
-/// ```
-#[must_use]
 pub struct GILGuard {
```

Comment:
Does `GILGuard` need to stay `pub` at all? Would `pub(crate)` or even private suffice if we use `EnsureGIL` elsewhere?

Reply:
Ah yes I nearly went for private and then forgot at the last second, will push... 😄